File: 914/x-of-a-kind-in-a-deck-of-cards.py
class Solution:
    def hasGroupsSizeX(self, deck):
        """
        :type deck: List[int]
        :rtype: bool
        """
        counts = {}
        for card in deck:
            if counts.get(card) is None:
                counts[card] = 1
            else:
                counts[card] += 1
            
        for i in range(2, len(deck)+1):
            if len(deck) % i:
                continue
            ok = True
            for item in counts.items():
                if item[1] % i:
                    ok = False
                    break
            if ok:
                return True
        return False

File: 914/x-of-a-kind-in-a-deck-of-cards.js
/**
 * @param {number[]} deck
 * @return {boolean}
 */
var hasGroupsSizeX = function (deck) {
    let counts = {};
    for (let card of deck) {
        counts[card] = ~~counts[card] + 1;
    }
    for (let i = 2; i <= deck.length; i++) {
        if (deck.length % i) continue;
        let ok = true;
        for (let item of Object.entries(counts)) {
            if (item[1] % i) {
                ok = false;
                break;
            }
        }
        if (ok) return true;
    }
    return false;
};